src/Control/Monad/Codensity.hs

```diff
@@ -47,9 +51,12 @@ import Data.Functor.Apply
 import Data.Functor.Kan.Ran
 import Data.Functor.Plus
 import Data.Functor.Rep
-#if __GLASGOW_HASKELL__ >= 708
+#if (__GLASGOW_HASKELL__ >= 708) && (__GLASGOW_HASKELL__ < 800)
 import Data.Typeable
 #endif
+#if __GLASGOW_HASKELL__ >= 802
+import GHC.Exts (TYPE)
```

Comment:
Levity polymorphism has been around since GHC 8.0—is there a reason you only enabled it on 8.2 or later?